bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingImpl.java

```diff
@@ -134,6 +135,18 @@ public void setLabel(@Nullable String label) {
         return Collections.unmodifiableList(new ArrayList<>(this.channels));
     }
 
+    @Override
+    public List<Channel> getChannelsOfGroup(String channelGroupId) {
+        List<Channel> channels = new ArrayList<>();
+        for (Channel channel : this.channels) {
+            ChannelUID channelUID = channel.getUID();
+            if (channelUID.isInGroup() && channelUID.getGroupId().equals(channelGroupId)) {
```

Comment:
The Eclipse Nullness Checker don't know that "getGroupId()" will return a non null value if "isInGroup()" returns true.
Why not simple using (I don't see any cons ATM):

```Java
if (channelGroupId.equals(channel.getUID().getGroupId())) {
    channels.add(channel);
}
```

Reply:
Done, thank for your review.